app/jobs/regular/auto_join_channel_batch.rb

```diff
@@ -30,11 +33,12 @@ def execute(args)
 
       new_member_ids = DB.query_single(create_memberships_query(category), query_args)
 
-      DB.exec(<<~SQL, channel_id: channel.id, joined: new_member_ids.size)
-        UPDATE chat_channels
-        SET user_count = user_count + :joined
-        WHERE id = :channel_id
-      SQL
+      # Only do this if we are running auto-join for a single user, if we
```

Comment:
Yes because we have a single user version of this here that's called by multiple things:

https://github.com/discourse/discourse-chat/blob/93e0cd8dcdfbff965357b2fbdaa2a271fd1bf556/lib/chat_channel_membership_manager.rb#L65-L72

I think the extra edge case of `auto_manage_channel_memberships` possibly only doing one user is fine here, shouldn't cause a double count because of the `user_count_stale` throttling